src/Nethermind/Nethermind.TxPool/Collections/DistinctValueSortedPool.cs

```diff
@@ -16,47 +16,53 @@
 
 using System;
 using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
 
 namespace Nethermind.TxPool.Collections
 {
-    public class DistinctValueSortedPool<TKey, TValue> : SortedPool<TKey, TValue>
+    public class DistinctValueSortedPool<TKey, TValue, TGroup> : SortedPool<TKey, TValue, TGroup>
```

Comment:
I suggest adding much more comments to this class. I do not get from name what thos class does / what kind of distinct value it sorts.